internal/pkg/dependencies/dependencies.go

```diff
@@ -0,0 +1,116 @@
+//
+// Example of difference between CLI and API dependencies implementations:
+//   - In the CLI the Storage API token is read from ENV or flag.
+//   - In the API the Storage API token is read from HTTP header.
+//
+// [Lazy] helper allows lazy initialization of a dependency on the first use.
+//
+// # Common Dependencies
+//
+// In this package is shared/common part of dependencies implementation:
+//   - [Base] interface contains basic dependencies (see [NewBaseDeps]).
+//   - [Public] interface contains dependencies available without authentication (see [NewPublicDeps]).
+//   - [Project] interface contains dependencies available after authentication (see [NewProjectDeps]).
+//   - [Mocked] interface provides dependencies mocked for tests (see [NewMockedDeps]).
+// [command design pattern]: https://refactoring.guru/design-patterns/command
+package dependencies
+
+import (
+	"context"
+
+	"github.com/jarcoal/httpmock"
+	"github.com/keboola/go-client/pkg/client"
+	"github.com/keboola/go-client/pkg/storageapi"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/cli/options"
+	"github.com/keboola/keboola-as-code/internal/pkg/env"
+	"github.com/keboola/keboola-as-code/internal/pkg/event"
+	"github.com/keboola/keboola-as-code/internal/pkg/filesystem"
+	"github.com/keboola/keboola-as-code/internal/pkg/log"
+	"github.com/keboola/keboola-as-code/internal/pkg/model"
+	projectPkg "github.com/keboola/keboola-as-code/internal/pkg/project"
+	"github.com/keboola/keboola-as-code/internal/pkg/state"
+	"github.com/keboola/keboola-as-code/internal/pkg/template"
+	"github.com/keboola/keboola-as-code/internal/pkg/template/repository"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/testproject"
+)
+
+// Base contains basic dependencies.
+type Base interface {
+	Envs() env.Provider
+	Logger() log.Logger
+	HttpClient() client.Client
+}
+
+// Public dependencies are available from the Storage API and other sources without authentication / Storage API token.
+type Public interface {
+	StorageApiHost() string
+	StorageApiPublicClient() client.Sender
+	StackFeatures() storageapi.FeaturesMap
+	StackServices() storageapi.ServicesMap
+	Components() *model.ComponentsMap
+	ComponentsProvider() *model.ComponentsProvider
+	EncryptionApiClient() client.Sender
+	Template(ctx context.Context, reference model.TemplateRef) (*template.Template, error)
+	TemplateRepository(ctx context.Context, reference model.TemplateRepository, forTemplate model.TemplateRef) (*repository.Repository, error)
+}
+
+// Project dependencies require authentication / Storage API token.
+type Project interface {
+	ProjectID() int
+	ProjectName() string
+	ProjectFeatures() storageapi.FeaturesMap
+	StorageApiTokenID() string
+	StorageApiClient() client.Sender
+	SchedulerApiClient() client.Sender
+	EventSender() event.Sender
+	ObjectIdGeneratorFactory() func(ctx context.Context) *storageapi.TicketProvider
```

Comment:
Tady by teda v názvu metody taky mělo být `ID` velký. (Jestli si dobře vybavuju z nějakýho toho Go style dokumentu že zkratky mají být velkýma písmenama?) Tak když už to přepisuješ, tak by bylo lepší to mít fixnutý. 🙂

Reply:
jj, dik, nevsimol som si, fixnem

Reply:
https://github.com/keboola/keboola-as-code/pull/782/commits/555b71f64bcd6595996cbba12ea19eb39fb59be9